docs/guides/upgrading-v3.md

```diff
@@ -0,0 +1,265 @@
+
+Finally, the `serverless studio` command has been removed: that feature was deprecated and is no longer available.
+
+### Service configuration
+
+The default Lambda runtime has changed from NodeJS 12 to NodeJS 14, given this is now the default runtime recommended by AWS.
+
+In `serverless.yml`, the `service` key no longer accepts a YAML object ([learn more](../deprecations.md#service-property-object-notation)).
+
+```yaml
+# Deprecated in v2, removed in v3:
+service:
+    name: my-service
+
+# Correct syntax:
+service: my-service
+```
+
+All options that used to be defined inside the `service` key have been moved to other sections (mentioned below in that document). This change clears up confusion that existed between the `service` and `provider` sections.
+
+CloudFormation extensions defined in `resources.extensions` are now validated: extensions that point to unknown resources will throw an error, which should help catch broken configuration before deploying ([learn more](../deprecations.md#defining-extensions-to-nonexistent-resources-in-resourcesextensions)).
+
+### API Gateway
+
+When configuring API Gateway, some configuration options have moved to a dedicated sub-section of `provider`. That will help clear up confusion with similar `httpApi` settings.
+
+```yaml
+provider:
+  # Deprecated in v2, removed in v3:
+  apiKeys: ...
+  resourcePolicy: ...
+  usagePlan: ...
+
+  # Correct syntax:
+  apiGateway:
+    apiKeys: ...
+    resourcePolicy: ...
+    usagePlan: ...
+```
+
+The `schema` option on HTTP events has also been renamed to `schemas`. That change allowed us to support much more schema validation features:
+
+```yaml
+functions:
+  hello:
+    handler: hello.handler
+    events:
+      - http:
+        ...
+        request:
+          # Deprecated in v2, removed in v3:
+          schema: ...
+          # Correct syntax:
+          schemas: ...
+```
+
+Learn more about [schema changes](../deprecations.md#aws-api-gateway-schemas).
+
+When an external API Gateway is used and imported via `provider.apiGateway.restApiId`, both `provider.logs.restApi` and `provider.tracing.apiGateway` options will be ignored. These settings are applicable only if the API Gateway is provisioned by Serverless Framework.
+
+The CloudFormation tags defined in `provider.tags` will now be correctly applied to HTTP APIs ([learn more](../deprecations.md#http-api-provider-tags)).
+
+Starting with v3, a bug fix for API Gateway authorizers will become the default behavior (to match AWS’ recommended behavior): `functions[].events[].http.authorizer.identitySource` will no longer be set to `method.request.header.Authorization` by default when caching is disabled (i.e. for authorizers of type "request" with `resultTtlInSeconds` set to "0") ([learn more](../deprecations.md#default-identitysource-for-httpauthorizer)).
```

Comment:
I would reword it to:

_Starting with v3 `functions[].events[].http.authorizer.identitySource` (setting related to API Gateway authorizers) will no longer be set to `method.request.header.Authorization` by default when caching is disabled_.

At least stating that _a bug fix for API Gateway authorizers will become the default behavior_ feels a bit weird to me

Reply:
We should also mention that it only applies to authorizers of `request` type with caching disabled, not in all cases when caching is disabled

Reply:
I've tried to push a change that still includes an explanation as to "why" the change was made (and removed the word "bugfix".

I also kept the details about "caching disabled".

Let me know if that's still correct.